deploy-agent/deployd/common/utils.py

```diff
@@ -236,7 +236,25 @@ def get_container_health_info(commit):
                         command = ['docker', 'inspect', '-f', '{{.State.Health.Status}}', name]
                         status = subprocess.run(command, check=True, stdout=subprocess.PIPE).stdout
                         if status:
-                            result.append(f"{name}:{status.decode().strip()}")
+                            status = status.decode().strip()
+                            if status == "unhealthy" and "redeploy_when_unhealthy=enabled" in parts[2]:
+                                labels = parts[2].split(',')
+                                max_retry = REDEPLOY_RETRY_TIMES
+                                for label in labels:
+                                    if "redeploy_retry_times" in label:
```

Comment:
let's get the retry into a helper function

Reply:
sure, thank you